packages/vega-expression/index.d.ts

```diff
@@ -1,21 +1,51 @@
-// TODO: add missing types
+/** Parse a JavaScript *expression* string and return the resulting abstract syntax tree in the ESTree format */
+export function parse(expression: string): object;
 
-export function parse(expression: string): any;
+/**
+ * Create a new output code generator configured according to the provided options:
+ * - `constants` — a hash of allowed top-level constant values
+ * - `functions` — a function that is given an AST visitor instance as input and returns an object of allowed functions
+ * - `blacklist` — an array of variable names that may not be referenced within the expression scope
+ * - `whitelist` — an array of variable names that may be referenced within the expression scope
+ * - `fieldvar` — the name of the primary data input argument within the generated expression function
+ * - `globalvar` — the name of the variable upon which to lookup global variables
+ */
+export function codegen(options: {
```

Comment:
Can you extract options to a new type (`CodegenOptions`) so we can remove the duplication between the docs below and the function comment?

Reply:
Done.

FYI,
- the options' explanations won't appear anymore when hoovering `codegen`;
- creating `CodegenOptions` wasn't necessary, we could have just removed the doc from the `codegen` function and leave the options where they were (as it is still now for the returned type of this function).